library/core/src/hint.rs

```diff
@@ -269,6 +269,16 @@ pub const unsafe fn assert_unchecked(cond: bool) {
 #[stable(feature = "renamed_spin_loop", since = "1.49.0")]
 pub fn spin_loop() {
     crate::cfg_select! {
+        miri => {
+            unsafe extern "Rust" {
+                safe fn miri_spin_loop();
+            }
+
+            // Miri does support some of the intrinsics that are called below, but to guarantee
+            // consistent behavior across targets, this custom function is used. To maximize the
+            // likelihood of finding actual issues, it will attempt to find another thread to run.
```

Comment:
```suggestion
            // consistent behavior across targets, this custom function is used.
```
What that function does is Miri's choice and maybe we'll change it in the future.